wsmaster/che-core-api-devfile/src/main/java/org/eclipse/che/api/devfile/server/DevfileConverter.java

```diff
@@ -121,6 +122,8 @@ public WorkspaceConfigImpl devFileToWorkspaceConfig(Devfile devfile)
         case PLUGIN_TOOL_TYPE:
           pluginsStringJoiner.add(tool.getId());
           break;
+        case KUBERNETES_TOOL_TYPE:
+          continue; // this kind of tools shouldn't be present in mapping
```

Comment:
throw exception?

Reply:
No, that's normal situation. We just have nothing to do with those tool type during conversion into WS config (and it should not be present in workspace plugins mapping attribute).

Reply:
Well, I also find this behavior and comment confusing. Maybe it would be better to rewrite this in another way like  the following:
```java
         case KUBERNETES_TOOL_TYPE:
         case OPENSHIFT_TOOL_TYPE:
           // this kind of tool contain only reference to tool configuration
           // it is ignored on converting to workspace config
          // then tool configuration should be provisioned to workspace config recipe
          // after downloading it.
           continue;                         
```

Reply:
Also, we are not going to support more than one tool that is not cheEditor nor chePlugin. Is there any validation check for it?